File: MyTestVueApp.Server/Entities/Comment.cs
namespace MyTestVueApp.Server.Entities
{ // ArtId, ArtName, ArtistId, Width, ArtLength, Encode, CreationDate, IsPublic
    public class Comment
    { // Art Table
        public int Id { get; set; }
        public int ArtistId { get; set; }
        public int ArtId { get; set; }
        public string Message { get; set; }
        public string CommenterName { get; set; }
        public DateTime CreationDate { get; set; }
        public bool Viewed { get; set; }

        //Represents the id that this comment is replying to
        public int ReplyId { get; set; }
        public bool CurrentUserIsOwner { get; set; } = false;
    }
}

File: mytestvueapp.client/src/entities/Comment.ts
export default class Comment {
  id?: number;
  artistId?: number;
  artId?: number;
  message?: string;
  commenterName?: string;
  creationDate: string;
  replyId?: number;
  currentUserIsOwner?: boolean;
  replies: Comment[];
  isViewed: boolean;

  constructor() {
    this.id = 0;
    this.artistId = 0;
    this.artId = 0;
    this.message = "";
    this.commenterName = "unknown";
    this.creationDate = "";
    this.replyId = 0;
    this.currentUserIsOwner = false;
    this.replies=[];
    this.isViewed=false;
  }
}
